src/mlpack/core/tree/rectangle_tree/rectangle_tree.hpp

```diff
@@ -214,6 +221,24 @@ class RectangleTree
    */
   ~RectangleTree();
 
+  /**
+   * Insert the given point into the tree and into the dataset held by the tree.
+   *
+   * An exception will be thrown if this is not called on the root of the tree.
+   */
+  template<typename InMatType>
+  void Insert(const InMatType& points);
+
+  /**
+   * Delete the given point index from the tree.  This will resize Dataset()
+   * accordingly and correct the point indexes in the rest of the tree, so is
+   * not a trivial operation.
+   *
+   * `pointIndex` must be between `0` and `numDescendants`, and an exception
+   * will be thrown if this is not called on the root of the tree.
+   */
+  void Delete(const size_t pointIndex);
```

Comment:
So what I understand is that we have just added the features that are making this tree unique.  and this did not exist in the past ?

Reply:
I was asking for Insert and Delete, I got it,